packages/core/utils/src/dml/helpers/entity-builder/build-indexes.ts

```diff
@@ -1,14 +1,15 @@
-import { EntityIndex } from "@medusajs/types"
+import { DMLSchema, EntityIndex } from "@medusajs/types"
 import { arrayDifference, isPresent } from "../../../common"
-import { DMLSchema } from "../../entity-builder"
 
 /*
   The DML provides an opinionated soft deletable entity as a part of every model
   We assume that deleted_at would be scoped in indexes in all cases as an index without the scope
   doesn't seem to be valid. If a case presents itself where one would like to remove the scope, 
   this will need to be updated to include that case.
 */
-export function transformIndexWhere(index: EntityIndex) {
+export function transformIndexWhere<TSchema extends DMLSchema>(
+  index: EntityIndex<TSchema>
+) {
   let where = index.where
```

Comment:
```suggestion
  let where = index.where + ''
```

Reply:
such that you can remove the check on where line 20

Reply:
and line 26 you can then change to `!where`

Reply:
will bring this to the base PR 🤞🏻